src/sage/rings/polynomial/multi_polynomial_sequence.py

```diff
@@ -849,7 +850,292 @@ def coefficient_matrix(self, sparse=True):
+            (
+            [0 0 0 0 1 0 0 0 2]
+            [0 1 0 0 0 0 0 2 0]
+            [1 0 0 0 0 0 2 0 0]
+            [0 0 0 0 0 1 0 1 0]
+            [0 0 1 0 0 1 0 0 0]
+            [0 1 0 0 1 0 0 0 0]
+            [0 0 0 0 0 0 1 0 0]
+            [0 0 0 0 1 0 0 0 0]
+            [0 0 0 1 0 0 0 0 0],
+            [(z, 0), (y, 0), (x, 0), (z, 1), (y, 1), (x, 1), (z, 2), (y, 2), (x, 2)],
+            [x^2*y, x*y^2, y^3, x^2*z, x*y*z, y^2*z, x*z^2, y*z^2, z^3]
+            )
+
+        Example in which we build rows using monomials that involve only a
+        subset of the ring variables (``variables=['x']``)::
+
+            sage: R.<x,y,z> = PolynomialRing(QQ)
+            sage: L = Sequence([2*y*z - 2*z^2 - 3*x + z - 3,
+            ....:               -3*y^2 + 3*y*z + 2*z^2 - 2*x - 2*y,
+            ....:               -2*y - z - 3])
+            sage: L.macaulay_matrix(1, variables=['x'], remove_zero=True, return_indices=True)
+            (
+            [ 0  0  0  0  0  0  0  0  0  2 -2 -3  0  1 -3]
+            [ 0  0  0  2 -2 -3  0  0  1  0  0 -3  0  0  0]
+            [ 0  0  0  0  0  0  0 -3  0  3  2 -2 -2  0  0]
+            [ 0 -3  0  3  2 -2 -2  0  0  0  0  0  0  0  0]
+            [ 0  0  0  0  0  0  0  0  0  0  0  0 -2 -1 -3]
+            [ 0  0  0  0  0  0 -2  0 -1  0  0 -3  0  0  0]
+            [-2  0 -1  0  0 -3  0  0  0  0  0  0  0  0  0],
+            [(1, 0), (x, 0), (1, 1), (x, 1), (1, 2), (x, 2), (x^2, 2)],
+            [x^2*y, x*y^2, x^2*z, x*y*z, x*z^2, x^2, x*y, y^2, x*z, y*z, z^2, x, y, z, 1]
+            )
+
+        TESTS::
+
+            sage: R.<x,y,z> = PolynomialRing(GF(7),order='deglex')
+            sage: from sage.rings.polynomial.multi_polynomial_sequence import PolynomialSequence_generic
+            sage: PolynomialSequence_generic([], R).macaulay_matrix(1)
+            Traceback (most recent call last):
+            ...
+            ValueError: the sequence of polynomials must be nonempty
+
+            sage: Sequence([x*y, x**2]).macaulay_matrix(-1)
+            Traceback (most recent call last):
+            ...
+            ValueError: the degree must be nonnegative
+
+            sage: Sequence([y*z + z^2 - 1,x*y - z^2 - x ]).macaulay_matrix(0, homogeneous=True)
+            Traceback (most recent call last):
+            ...
+            ValueError: all the polynomials of the sequence must be homogeneous
+
+            sage:Sequence([y*z + z^2 - 1,x*y - z^2 - x ]).macaulay_matrix(0, row_order="POP")
+            Traceback (most recent call last):
+            ...
+            ValueError: the argument of ``row_order`` must be ``None``, "TOP" or "POT"
+
+            sage: R1.<t>=PolynomialRing(GF(3))
+            sage: Sequence([y*z + z^2 - 1,x*y - z^2 - x ]).macaulay_matrix(1, variables=[t])
+            Traceback (most recent call last):
+            ...
+            ValueError: the variables must be in the polynomial ring
+
```

Comment:
```suggestion

        sage: Sequence([y*z + z^2 - 1,x*y - z^2 - x ]).macaulay_matrix(1, row_order="increasing")
        Traceback (most recent call last):
        ...
        ValueError: the argument of ``row_order`` must be ``None``, "TOP" or "POT"

        sage: R.<x,y,z> = PolynomialRing(GF(7),order='degrevlex(2),neglex(1)')
        sage: Sequence([y*z + z^2 - 1,x*y - z^2 - x ]).macaulay_matrix(1, variables=[x,z])
        Traceback (most recent call last):
        ...
        ValueError: impossible to use the original term order (most likely because it was a block order). Please specify the term order for the subring

```

(for test code coverage)